x-pack/plugins/index_management/__jest__/client_integration/helpers/http_requests.ts

```diff
@@ -17,139 +18,105 @@ export interface ResponseError {
 }
 
 // Register helpers to mock HTTP Requests
-const registerHttpRequestMockHelpers = (server: SinonFakeServer) => {
-  const setLoadTemplatesResponse = (response: HttpResponse = []) => {
-    server.respondWith('GET', `${API_BASE_PATH}/index_templates`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setLoadIndicesResponse = (response: HttpResponse = []) => {
-    server.respondWith('GET', `${API_BASE_PATH}/indices`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setReloadIndicesResponse = (response: HttpResponse = []) => {
-    server.respondWith('POST', `${API_BASE_PATH}/indices/reload`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setLoadDataStreamsResponse = (response: HttpResponse = []) => {
-    server.respondWith('GET', `${API_BASE_PATH}/data_streams`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setLoadDataStreamResponse = (response: HttpResponse = []) => {
-    server.respondWith('GET', `${API_BASE_PATH}/data_streams/:id`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setDeleteDataStreamResponse = (response: HttpResponse = []) => {
-    server.respondWith('POST', `${API_BASE_PATH}/delete_data_streams`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setDeleteTemplateResponse = (response: HttpResponse = []) => {
-    server.respondWith('POST', `${API_BASE_PATH}/delete_index_templates`, [
-      200,
-      { 'Content-Type': 'application/json' },
-      JSON.stringify(response),
-    ]);
-  };
-
-  const setLoadTemplateResponse = (response?: HttpResponse, error?: any) => {
-    const status = error ? error.status || 400 : 200;
-    const body = error ? error.body : response;
-
-    server.respondWith('GET', `${API_BASE_PATH}/index_templates/:id`, [
```

Comment:
With sinon fake server we could use wildcards for url params, that is no longer allowed with HttpSetup mock. So we need to directly specify the full url when mocking api endpoints.  For this particular case would look something like:

```typescript
  const setLoadTemplateResponse = (
    templateId: string,
    response?: HttpResponse,
    error?: ResponseError
  ) => mockResponse('GET', `${API_BASE_PATH}/index_templates/${templateId}`, response, error);
```